extensions/ql-vscode/src/mocks/mock-gh-api-server.ts

```diff
@@ -33,8 +49,98 @@ export class MockGitHubApiServer {
     // TODO: Implement logic to list all available scenarios.
   }
 
-  public recordScenario(): void {
-    // TODO: Implement logic to record a new scenario to a directory.
+  public async recordScenario(): Promise<void> {
+    if (this.recorder.isRecording) {
+      void window.showErrorMessage('A scenario is already being recorded. Use the "Save Scenario" or "Cancel Scenario" commands to finish recording.');
+      return;
+    }
+
+    this.recorder.start();
+    await commands.executeCommand('setContext', 'codeQL.mockGitHubApiServer.recording', true);
+
+    await window.showInformationMessage('Recording scenario. To save the scenario, use the "CodeQL Mock GitHub API Server: Save Scenario" command.');
+  }
+
+  public async saveScenario(): Promise<void> {
+    const scenariosDirectory = await this.getScenariosDirectory();
+    if (!scenariosDirectory) {
+      return;
+    }
+
+    await commands.executeCommand('setContext', 'codeQL.mockGitHubApiServer.recording', false);
```

Comment:
Let's add a comment to explain why we're doing this as it's not immediately obvious. E.g. `set a value in the context to track whether we're recording so we can decide whether commands around recording should be visible`